"Unknown repository" toast notification when using private git npm dependencies

### Bug description

👋  Hey there folks! We've noticed that when we install npm dependencies for one of our applications there's a toast message thrown in Gitpod's VS Code IDE alerting about an unknown repository. It looks like this:
![image](https://user-images.githubusercontent.com/6409227/150246029-0f42cd44-5ed9-4ac9-a130-bbba3d293f91.png)

We've simply been ignoring the message since everything does install correctly. I recorded a Loom for you to review that recreates the issue with a minimal set-up: https://www.loom.com/share/66ebf0ae586e4dd0921b6fbc2aaaf3bf

### Steps to reproduce

1. Create a private git repository for an npm package
2. Add this new package as a dependency to a Gitpod enabled repository using the `github:` protocol helper
```
...
"devDependencies": {
    "private-package": "github:joepurdy/private-npm-package"
}
...
```
3. Run `npm install` from within Gitpod to trigger the "unknown repository" error

### Workspace affected

_No response_

### Expected behavior

Ideally no toast warning would occur in this case since there's no real issue. 

I think where the issue could become thorny is if a user doesn't actually have access to the private repository that is specified as a dependency. In this case the packages would fail to install, but regardless it's not really a problem for Gitpod to be concerned with since lack of access at the user level is the responsibility of whoever manages user access to the associated git repositories.

### Example repository

Here's a snapshot to the demo repo from the Loom recording: https://gitpod.io#snapshot/00ac0dd4-e05b-4e7d-88b8-af19c606ad9a

Note: this should still recreate the toast notification, but you'll also see an error like this in the terminal output:
```
npm ERR! code 128
npm ERR! An unknown git error occurred
npm ERR! command git --no-replace-objects ls-remote ssh://git@github.com/joepurdy/private-npm-package.git
```

That's expected behavior since you won't have access to the private dependency's source repo. This is what I meant about it not really being a problem for Gitpod to solve for in the event a user lacks the permissions. In that case there's already a clear error from npm about lacking the right access.

### Anything else?

/cc @loujaybee I imagine this might be interesting to the IDE team 😁

--- a/components/gitpod-cli/cmd/credential-helper.go
+++ b/components/gitpod-cli/cmd/credential-helper.go
@@ -54,10 +54,13 @@ var credentialHelper = &cobra.Command{
 			fmt.Printf("username=%s\npassword=%s\n", user, token)
 		}()
 
-		repoURL, gitCommand := parseProcessTree()
+		gitCommandData := parseProcessTree()
+		if gitCommandData == nil {
+			return
+		}
 
 		// Starts another process which tracks the executed git event
-		gitCommandTracker := exec.Command("/proc/self/exe", "git-track-command", "--gitCommand", gitCommand)
+		gitCommandTracker := exec.Command("/proc/self/exe", "git-track-command", "--gitCommand", gitCommandData.GitCommand)
 		err = gitCommandTracker.Start()
 		if err != nil {
 			log.WithError(err).Print("error spawning tracker")
@@ -95,7 +98,7 @@ var credentialHelper = &cobra.Command{
 
 		validator := exec.Command("/proc/self/exe", "git-token-validator",
 			"--user", resp.User, "--token", resp.Token, "--scopes", strings.Join(resp.Scope, ","),
-			"--host", host, "--repoURL", repoURL, "--gitCommand", gitCommand)
+			"--host", host, "--repoURL", gitCommandData.RepoUrl, "--gitCommand", gitCommandData.GitCommand)
 		err = validator.Start()
 		if err != nil {
 			log.WithError(err).Print("error spawning validator")
@@ -131,7 +134,10 @@ func parseHostFromStdin() string {
 	return host
 }
 
-func parseProcessTree() (repoUrl string, gitCommand string) {
+func parseProcessTree() (result *struct {
+	RepoUrl    string
+	GitCommand string
+}) {
 	gitCommandRegExp := regexp.MustCompile(`git(,\d+\s+|\s+)(push|clone|fetch|pull|diff)`)
 	repoUrlRegExp := regexp.MustCompile(`\sorigin\s*(https:[^\s]*)\s`)
 	pid := os.Getpid()
@@ -141,24 +147,33 @@ func parseProcessTree() (repoUrl string, gitCommand string) {
 			return
 		}
 		cmdLineString := strings.ReplaceAll(cmdLine, string(byte(0)), " ")
-		if gitCommand == "" {
-			match := gitCommandRegExp.FindStringSubmatch(cmdLineString)
-			if len(match) == 3 {
-				gitCommand = match[2]
+
+		var gitCommand string
+		matchCommand := gitCommandRegExp.FindStringSubmatch(cmdLineString)
+		if len(matchCommand) == 3 {
+			gitCommand = matchCommand[2]
+		}
+
+		var repoUrl string
+		matchRepo := repoUrlRegExp.FindStringSubmatch(cmdLineString)
+		if len(matchRepo) == 2 {
+			repoUrl = matchRepo[1]
+			if !strings.HasSuffix(repoUrl, ".git") {
+				repoUrl = repoUrl + ".git"
 			}
 		}
-		if repoUrl == "" {
-			match := repoUrlRegExp.FindStringSubmatch(cmdLineString)
-			if len(match) == 2 {
-				repoUrl = match[1]
-				if !strings.HasSuffix(repoUrl, ".git") {
-					repoUrl = repoUrl + ".git"
-				}
-			}
-		}
+
 		if repoUrl != "" && gitCommand != "" {
+			result = &struct {
+				RepoUrl    string
+				GitCommand string
+			}{
+				RepoUrl:    repoUrl,
+				GitCommand: gitCommand,
+			}
 			return
 		}
+
 		statsString := readProc(pid, "stat")
 		if statsString == "" {
 			return

